src/components/comments/CommentForm.jsx

```diff
@@ -1,9 +1,60 @@
+import { useState } from "react";
+import { postComment } from "../../utils/api";
+import SuccessDisplay from "../minor/SuccessDisplay";
+import ErrorDisplay from "../minor/ErrorDisplay";
+
 const CommentForm = ({ article_id, addComment }) => {
-  const handleSubmit = (e) => {
-    //posting goes here
+  const [body, setBody] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [error, setError] = useState(null);
+  const [success, setSuccess] = useState(false);
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (!body) {
+      setError("Comment cannot be empty");
+      return;
+    }
+    setIsSubmitting(true);
+    setError(null);
+    setSuccess(false);
+
+    try {
+      const newComment = await postComment(article_id, "jessjelly", body);
+      addComment(newComment);
+      setBody("");
+      setSuccess(true);
+      setTimeout(() => setSuccess(false), 3000);
```

Comment:
What is the purpose of this setTimeout?

Reply:
It's for the "Comment posted successfully" message, so it doesn't stay on the screen permanently.